### PR TITLE
lora: compile drivers based on devicetree 

### PR DESCRIPTION
**Pull request by JordanYates**
Similar to the changes made in https://github.com/zephyrproject-rtos/zephyr/pull/48048, compile in the appropriate lora
modem driver when `CONFIG_LORA=y`. LoRa and LoRaWAN subsystems are still
marked as unstable so no additional deprecation cycle is needed.

Signed-off-by: Jordan Yates <jordan.yates@data61.csiro.au>

### CODE REVIEW

**Comment by stephanosio** [comment]
Please rebase to fix the manifest and assigner workflow failures.

File: drivers/lora/Kconfig.sx12xx [hunk 1]
```diff
@@ -4,49 +4,30 @@
...
-	  is discovered from the device tree and should be correct for
-	  most users.
-
 config LORA_SX127X
 	bool "Semtech SX127x driver"
+	default y
 	depends on DT_HAS_SEMTECH_SX1272_ENABLED || DT_HAS_SEMTECH_SX1276_ENABLED
 	select HAS_SEMTECH_SX1272 if DT_HAS_SEMTECH_SX1272_ENABLED
 	select HAS_SEMTECH_SX1276 if DT_HAS_SEMTECH_SX1276_ENABLED
+	depends on SPI
```
Comment (nixward) [inline drivers/lora/Kconfig.sx12xx:13]: Should these be:
```suggestion
	select SPI
	select GPIO
```

Then the `CONFIG_SPI=y` and `CONFIG_GPIO=y` lines can also be potentially removed from `prj.conf` files.

Comment (JordanYates) [reply drivers/lora/Kconfig.sx12xx:13]: That is potentially another change to make, but I would be looking to do a complete pass of all usages of `depends on SPI` to clean up inconsistencies. It's not the primary purpose of this PR.

File: samples/drivers/lora/receive/prj.conf [hunk 2]
```diff
@@ -2,5 +2,4 @@ CONFIG_LOG=y
 CONFIG_SPI=y
 CONFIG_GPIO=y
 CONFIG_LORA=y
-CONFIG_LORA_SX12XX=y
```
Comment (mniestroj) [inline samples/drivers/lora/receive/prj.conf:5]: This should probably be squashed to previous commit in order to preserve bisectability.

**Review by mniestroj — COMMENTED** [review]
IMO those 2 commits should be squashed for bisectability. Otherwise LGTM.

File: samples/drivers/lora/receive/prj.conf (hunk 2 above)
Comment (JordanYates) [reply samples/drivers/lora/receive/prj.conf:5]: Done

**Review by mniestroj — APPROVED** [review]

**Review by Mani-Sadhasivam — APPROVED** [review]

**Pull request merged**